kmip/core/messages/payloads/mac.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2017 Pure Storage, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from kmip.core import attributes
+from kmip.core import enums
+from kmip.core.enums import Tags
+
+from kmip.core.objects import Data, MACData
+
+from kmip.core.primitives import Struct
+
+from kmip.core.utils import BytearrayStream
+
+
+# 4.33
+class MACRequestPayload(Struct):
+
+    def __init__(self,
+                 unique_identifier=None,
+                 cryptographic_parameters=None,
+                 data=None):
+        super(MACRequestPayload, self).__init__(
+            tag=enums.Tags.REQUEST_PAYLOAD)
+        self.unique_identifier = unique_identifier
+        self.cryptographic_parameters = cryptographic_parameters
+        self.data = data
```

Comment:
Good eye. The answer is "sort of." If you look through the library you'll find lots of places where parse errors can occur, raising exceptions. This is pretty standard in OOP. The problem is that this makes the code harder to work with. For example, the user/client may provide an extra unneeded attribute value, or instead omit one, when making a client call. This mistake may not trigger an exception until way down the processing stack when reading or writing an entire request/response structure. This is frustrating to the user because they get an error related to library internals that they don't understand, and its frustrating for developers because the source of the error is far away from where the error is actually detected, making it hard to find and fix.

There are a couple of solutions to this. The obvious one is to just put in better error handling. The problem is this would pretty much have to happen everywhere and it would even further bloat and complicate the code. An alternative (which is what I explored for a time) was to rethink what is actually a parse error. The idea here is that if a request/response message can be parsed successfully (i.e., the fields present are formatted properly and there are no partial fields that can't get read), let it get parsed. If the payload is missing certain fields, let the user/client know by a more organic, less intrusive mechanism than just erroring out. I started playing around with this with the `GetAttributeListResponsePayload`, to see if I liked how it felt/worked and if it caused any problems with active deployments. I don't have enough "data" yet to determine if it's a viable way forward, so I'm not sure if I'll stick with it.

For now, with the MAC payloads, I'm fine with you putting in those read/write exceptions for missing required fields. If need be, I'll patch in the right exception(s) needed for `GetAttributeList` and make everything consistent again.